plugins/modules/zabbix_user_role.py

```diff
@@ -0,0 +1,183 @@
+    server_url: http://zabbix.scz-vm.net/
+    login_user: username
+    login_password: login_password
+    state: present
+    name: Operators
+    type: 1
+    rules:
+      ui.default_access: 0
+      ui:
+        - name: "monitoring.hosts"
+          status: 0
+        - name: "monitoring.maps"
+          status: 1
+'''
+
+RETURN = r'''
+# Return values
+msg:
+    description: The result of the action
+    type: str
+    returned: always
+    sample: 'No action'
+changed:
+    description: The consequence of the action
+    type: bool
+    returned: always
+    sample: False
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.zabbix.plugins.module_utils.version import LooseVersion
+
+from ansible_collections.community.zabbix.plugins.module_utils.base import ZabbixBase
+import ansible_collections.community.zabbix.plugins.module_utils.helpers as zabbix_utils
+
+
+class UserRole(ZabbixBase):
+    def __find_val(self, outval, inval):
+        if outval == str(inval):
+            return True
+        return False
+
+    def __find_list(self, outval, inval):
+        if set(outval) == set(inval):
+            return True
+        return False
+
+    def __find_dict(self, outval, inval):
+        for out in outval:
+            m = True
+            for k, v in inval.items():
+                if out[k] == str(v):
+                    continue
+                else:
+                    m = False
+            if m:
+                break
+        return m
+
+    def is_part_of(self, inp, out):
+        verdict = True
+        for rule, value in inp.items():
+            if not isinstance(value, list):
+                verdict = verdict and self.__find_val(out.get(rule, ''), value)
+            else:
+                if len(value):
+                    if not isinstance(value[0], dict):
+                        verdict = verdict and self.__find_list(out.get(rule, []), value)
+                    else:
+                        for v in value:
+                            verdict = verdict and self.__find_dict(out.get(rule, {}), v)
+                else:
+                    verdict = verdict and self.__find_list(rule, value)
+        return verdict
+
+    def get_user_role(self, name):
+        result = self._zapi.role.get({
+            "output": "extend",
+            "selectRules": "extend",
+            "filter": {"name": name}
+        })
+        return result
+
+
+def main():
+    msg = "No action"
+    changed = False
+
+    argument_spec = zabbix_utils.zabbix_common_argument_spec()
+    argument_spec.update(dict(
+        state=dict(type='str', required=False, default='present'),
+        name=dict(type='str', required=True),
+        type=dict(type='int', required=False, default=1),
+        rules=dict(type='dict', required=True),
+    ))
+
+    # the AnsibleModule object
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
```

Comment:
Something like this https://github.com/ansible-collections/community.zabbix/blob/10a4040a8a88d50068d3ecc8475f877db8603864/plugins/modules/zabbix_mediatype.py#L718-L722 is missing here